core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -982,6 +982,7 @@ private Object readResolve() {
         if (jdks == null) {
             jdks = new ArrayList<>();
         }
+        slaveAgentPort = SystemProperties.getInteger(Jenkins.class.getName()+".slaveAgentPort", slaveAgentPort);
```

Comment:
Then this should have been the behavior from the start. Which [it is neither in behavior nor intent](https://github.com/jenkinsci/jenkins/commit/653fbdb65024b1b528e21f682172885f7111bba9#commitcomment-17342255), and it's [documented differently as well](https://wiki.jenkins-ci.org/display/JENKINS/Features+controlled+by+system+properties).

Or are you claiming that there is absolutely nobody who needed a fixed port for the initial setup, but changed it on the UI later?

The system property was unfortunately not named `initialPort` or similar due to @kohsuke circumventing the usual PR process, but it's too late to change that.

My proposal if you want to have different behavior, do it properly: New system property, called `enforcePort` (or similar), that removes the option from the UI, skips setting the option on form submission, and `readResolve`s.